plugin/tmux_navigator.vim

```diff
@@ -60,7 +60,8 @@ function! s:TmuxAwareNavigate(direction)
   " a) we're toggling between the last tmux pane;
   " b) we tried switching windows in vim but it didn't have effect.
   if tmux_last_pane || nr == winnr()
-    if g:tmux_navigator_save_on_switch
+    " Unly save the file if it has a filename
+    if g:tmux_navigator_save_on_switch && @% != ""
```

Comment:
Could you extract the non-file check to a function? Something like `s:isFileBacked()` or similar?

Also, is there any other hook that might be more explicit to check if the buffer is backed by a file? Something like `buftype`?